### ISSUE TITLE
Update not reflect 

### ISSUE THREAD

**Issue by chingthampiyarilal** [wix/react-native-calendars#2464]
Please make our job easier by filling this template out to completion. If you're requesting a feature instead of reporting a bug, please feel free to skip the Environment and Reproducible Demo sections.

## Description

1-2 sentences describing the problem you're having or the feature you'd like to request.

In Agenda Calender,I creating different event on same date if i tried to update the each event everything is reflected on the data but not the UI.

## Expected Behavior

What action did you perform, and what did you expect to happen?

I change to delete inside the package module reservation.js "shouldComponentUpdate(nextProps) {....."          ===> When i change this part it works fine
![a](https://github.com/wix/react-native-calendars/assets/158568104/a03198c2-7f0f-484c-85ce-82e5dfedb3a4)
![b](https://github.com/wix/react-native-calendars/assets/158568104/438557ec-46e3-43bd-85b4-e3811817740a)

## Observed Behavior

What actually happened when you performed the above actions?

Updating event not reflected in the UI but all are reflected in data/ logs..

If there's an error message, please paste the *full terminal output and error message* in this code block:

```I have an warning not error  : VirtualizedList: You have a large list that is slow to update - make sure your renderItem function renders components that follow React performance best practices like PureComponent, shouldComponentUpdate, etc. {"contentLength": 2948, "dt": 843, "prevDt": 883}
!
```

## Environment

Please run these commands in the project folder and fill in their results:

* `npm ls react-native-calendars`:
* `npm ls react-native`:

Also specify:

1. Device/emulator/simulator & OS version:

## Reproducible Demo

Please provide a minimized reproducible demonstration of the problem you're reporting.

Issues that come with minimal repro's are resolved much more quickly than issues where a maintainer has to reproduce themselves.

## Screenshots

Screenshots or gifs of the issue and the suggested fix will help us move faster with the review process.

**Comment by srinu6** [comment]
I am also facing the same issue, were you able to fix this issue? @chingthampiyarilal

**Comment by chingthampiyarilal** [comment]
@srinu6, It works when i delete "shouldComponentUpdate(nextProps) {....."  function,   inside the  node_modules\react-native-calendars\src\agenda\reservation-list\reservation.js

**Comment by hughyyyy** [comment]
Yes, I can confirm, shouldComponentUpdate is not working properly.

**Comment by jamesa3** [comment]
This issue is happening for me as well. It seems to be caused by:
```
if (!d1 && !d2) {
  changed = false;
}
```
in `node_modules\react-native-calendars\src\agenda\reservation-list\reservation.js` `shouldComponentUpdate`.

The `d1` and `d2` props are only provided to the first item on a given date, so any items after the first item on a date don't get updated properly.

**Comment by livnunes** [comment]
This is definitely still an issue and shouldn't be closed.
With the following code
`
shouldComponentUpdate(nextProps) {
	const d1 = this.props.date;
	const d2 = nextProps.date;
	const r1 = this.props.item;
	const r2 = nextProps.item;

	// If both dates are null/undefined, check items
	if (!d1 && !d2) {
		return this.props.rowHasChanged ? this.props.rowHasChanged(r1, r2) : r1 !== r2;
	}

	// If both dates are defined, compare dates
	if (d1 && d2 && d1.getTime() !== d2.getTime()) {
		return true;
	}

	// If one of the dates is null/undefined, consider it changed
	if (!!d1 !== !!d2) {
		return true;
	}

	// If dates are the same or null, check items
	if (this.props.rowHasChanged) {
		return this.props.rowHasChanged(r1, r2);
	}

	// Default item comparison if rowHasChanged is not defined
	return r1 !== r2;
}
`

I managed to make it work but also had to add `rowHasChanged={(r1, r2) => JSON.stringify(r1) !== JSON.stringify(r2)}` to the <Agenda> component.
I don't know if this is remotely close to being a good idea to do but for now at least from my tests it seemed to work when I update any of the events in a specific day.

**Comment by chux0519** [comment]
> This is definitely still an issue and shouldn't be closed. With the following code
> 
> ```
> shouldComponentUpdate(nextProps) {
> 	const d1 = this.props.date;
> 	const d2 = nextProps.date;
> 	const r1 = this.props.item;
> 	const r2 = nextProps.item;
> 	// If both dates are null/undefined, check items
> 	if (!d1 && !d2) {
> 		return this.props.rowHasChanged ? this.props.rowHasChanged(r1, r2) : r1 !== r2;
> 	}
> 	// If both dates are defined, compare dates
> 	if (d1 && d2 && d1.getTime() !== d2.getTime()) {
> 		return true;
> 	}
> 	// If one of the dates is null/undefined, consider it changed
> 	if (!!d1 !== !!d2) {
> 		return true;
> 	}
> 	// If dates are the same or null, check items
> 	if (this.props.rowHasChanged) {
> 		return this.props.rowHasChanged(r1, r2);
> 	}
> 	// Default item comparison if rowHasChanged is not defined
> 	return r1 !== r2;
> }
> ```
> 
> I managed to make it work but also had to add `rowHasChanged={(r1, r2) => JSON.stringify(r1) !== JSON.stringify(r2)}` to the Agenda component. I don't know if this is remotely close to being a good idea to do but for now at least from my tests it seemed to work when I update any of the events in a specific day.

encountered the same problem, and this solution solved
